### ISSUE TITLE
frontend: predicate push-down doesn't work

### ISSUE THREAD

**Issue by fuyufjh** [singularity-data/risingwave#2463]
In the following plan, `BatchFilter { predicate: ($1 < 10:Int32) }` should be transposed through Join

```sql
create table t1 (v1 int not null, v2 int not null);
create table t2 (v1 int not null, v2 int not null);
```

```
eric=> explain select t1.v2 as t1_v2, t2.v2 as t2_v2 from t1 join t2 on t1.v1 = t2.v1 and t1.v2 < 10;
                           QUERY PLAN
----------------------------------------------------------------
 BatchExchange { order: [], dist: Single }
   BatchProject { exprs: [$1, $3], expr_alias: [t1_v2, t2_v2] }
     BatchFilter { predicate: ($1 < 10:Int32) }
       BatchHashJoin { type: Inner, predicate: $0 = $2 }
         BatchExchange { order: [], dist: HashShard([0]) }
           BatchScan { table: t1, columns: [v1, v2] }
         BatchExchange { order: [], dist: HashShard([0]) }
           BatchScan { table: t2, columns: [v1, v2] }
```

**Comment by likg227** [comment]
```
  optimized_logical_plan: |
    LogicalProject { exprs: [$1, $3] }
      LogicalJoin { type: Inner, on: ($0 = $2) AND ($1 < 10:Int32) }
        LogicalScan { table: t1, columns: [x, y] }
        LogicalScan { table: t2, columns: [x, y] 
```
Because we use FilterJoin to push predicates down and there is no LogicalFilter here, we can't push the predicates in `on` of LogicalJoin down.

**Comment by likg227** [comment]
I'm planning to implement predicates pushdown in a way like prun_col, and I think the new method can deal with this case.

**Comment by xxchan** [comment]
I propose a simple fix in #2516
